From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: DerEchtePilz <81232921+DerEchtePilz@users.noreply.github.com>
Date: Tue, 20 Aug 2024 12:38:20 +0200
Subject: [PATCH] Fix issues with CompassMeta

adds a clearLodestone method to reset the
compass to a normal one.

diff --git a/src/main/java/org/bukkit/inventory/meta/CompassMeta.java b/src/main/java/org/bukkit/inventory/meta/CompassMeta.java
index 5040ab6190b41442986d2a734a8e782df0eab2f6..ec7018e7d04d3fcc4d177235636e82fa7b714ec1 100644
--- a/src/main/java/org/bukkit/inventory/meta/CompassMeta.java
+++ b/src/main/java/org/bukkit/inventory/meta/CompassMeta.java
@@ -28,7 +28,7 @@ public interface CompassMeta extends ItemMeta {
     /**
      * Sets the location this lodestone compass will point to.
      *
-     * @param lodestone new location or null to clear
+     * @param lodestone new location or null to clear the targeted location
      */
     void setLodestone(@Nullable Location lodestone);
 
@@ -52,6 +52,14 @@ public interface CompassMeta extends ItemMeta {
      */
     void setLodestoneTracked(boolean tracked);
 
+    // Paper start - Add method to reset Lodestone compasses back to normal ones
+    /**
+     * Reset this compass to a normal compass, removing any tracked
+     * location.
+     */
+    void clearLodestone();
+    // Paper end - Add method to reset Lodestone compasses back to normal ones
+
     @Override
     CompassMeta clone();
 }
